### ISSUE TITLE
Preserve multiple HTTP header values as a list instead of a joined string

### ISSUE THREAD

**Issue by aabmass** [open-telemetry/opentelemetry-python-contrib#2360]
Right now, HTTP headers are stored as list type span attribute with one item. If there are repeated headers, they are joined with a comma as (being implemented in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2266). We could instead preserve each header value as a separate string.

Current behavior:
```
{
  "http.request.header.custom_test_header_1": ("test-header-value-1,test-header-value-2",),
}
```
Proposed behavior
```
{
  "http.request.header.custom_test_header_1": ("test-header-value-1", "test-header-value-2"),
}
```

This may require changing some public signatures or adding new overloads to keep backward compatibility.

**Comment by aabmass** [comment]
Somewhat related, a bunch of signatures in ASGI and WSGI instrumentations are already incorrect https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2358

**Comment by samuelcolvin** [comment]
Wait, I think I was making this change in #2266?

**Comment by samuelcolvin** [comment]
Fix proposed in #2361.
